Proxy

Krokop opened the pull request:
Додав view, tests, та конфігурації для запуску як окремого standalone.

Krokop committed:
add proxy, with tests

Krokop committed:
fix tests

IrynaPomazan committed:
Renamed field's name in file. Return yaml if exists 'Accept' header

IrynaPomazan committed:
Rename create_file to prepare data. Formatting.

Krokop committed:
Merge pull request #24 from ITVaan/change_fields

Renamed field's name in file. Return yaml if exists 'Accept' header

IrynaPomazan committed:
add default value for optional fields

Krokop committed:
Merge pull request #26 from ITVaan/fix_bug

add default value for optional fields

Krokop committed:
megre with add_edr_proxy

kroman0 commented on openprocurement/integrations/edr/traversal.py:
```diff
@@ -0,0 +1,27 @@
+        self.request = request
+
+
+def factory(request):
```
Для proxy не потрібно factory оскільки в аплікації не передбачено контекст.

kroman0 replied:
https://github.com/openprocurement/openprocurement.documentservice/blob/master/openprocurement/documentservice/views.py#L29

Krokop replied:
прибрав factory

kroman0 commented on openprocurement/integrations/edr/__init__.py:
```diff
@@ -0,0 +1,70 @@
+    config = Configurator(
+        autocommit=True,
+        settings=settings,
+        authentication_policy=AuthenticationPolicy(settings['auth.file'], __name__),
```
Для proxy потрібно Basic Authentication з використанням імені і пароля користувача - https://github.com/openprocurement/openprocurement.documentservice/blob/master/openprocurement/documentservice/views.py#L29

Krokop replied:
переніс Authentication з document service.

kroman0 commented on openprocurement/integrations/edr/traversal.py:
```diff
@@ -0,0 +1,27 @@
+class Root(object):
+    __name__ = None
+    __parent__ = None
+    __acl__ = [
```
потрібен поділ повноважень для майданчиків і бота(моста)

Krokop replied:
розділив

Krokop committed:
update roles, remove prefix for url, change verify view from class to function, remove factory, remove travesal.py,
add Root to utils, scan all user from auth.ini to module variable.
a lot of refactoring.

Krokop commented:
Прибрав префікс, додав сканування всіх юзерів. Потрібен ревью

kroman0 commented on openprocurement/integrations/edr/__init__.py:
```diff
@@ -0,0 +1,70 @@
+
+
+LOGGER = getLogger("{}.init".format(__name__))
+SECURITY = {u'admins': {u'names': [], u'roles': ['_admin']}, u'members': {u'names': [], u'roles': ['_admin']}}
```
Його тут не треба

Krokop replied:
прибрав

kroman0 commented on openprocurement/integrations/edr/auth.py:
```diff
@@ -0,0 +1,124 @@
+from ConfigParser import ConfigParser
+
+
+class AuthenticationPolicy(BasicAuthAuthenticationPolicy):
```
AuthenticationPolicy вже не використовуєть

Krokop replied:
прибрав

kroman0 commented on openprocurement/integrations/edr/__init__.py:
```diff
@@ -0,0 +1,70 @@
+    config.add_subscriber(set_renderer, NewRequest)
+    config.add_subscriber(beforerender, BeforeRender)
+
+    config.registry.server_id = settings.get('id', '')
```
server_id, health_threshold і update_after не використовуються

Krokop replied:
прибрав

kroman0 commented on openprocurement/integrations/edr/utils.py:
```diff
@@ -0,0 +1,200 @@
+        ]
+
+
+def beforerender(event):
```
цей beforerender потрібен тільки для op.api

Krokop replied:
прибрав

kroman0 commented on openprocurement/integrations/edr/utils.py:
```diff
@@ -0,0 +1,200 @@
+    __parent__ = None
+    __acl__ = [
+        (Allow, 'g:platforms', 'verify'),
+        (Allow, 'g:robots', 'verify'),
```
Роботу має надаватись розширений витяг з реєстру, тому доцільно додати g:robots додаткові повноваження, які будуть перевірятись в verify_user.

Krokop replied:
додав додав додаткові повноваження

kroman0 reviewed (changes requested):
Почистіть код від зайвого

Krokop committed:
clean code

Krokop commented:
Потрібен ревью

Krokop committed:
add yaml to requires in setup

kroman0 reviewed (changes requested):
Є части на коду, яка не використовується або не тестується, тести можна виключити з coverage - 
https://coveralls.io/github/openprocurement/openprocurement.integrations.edr. Потрібно прибрати все що потребує контексту чи згадки про тендери

Krokop committed:
add coveragerc

Krokop committed:
fix bug, and update tests

Krokop committed:
remove duplicates

Krokop committed:
add test for forbidden

Krokop committed:
forgot about status

Krokop committed:
fix test

kroman0 commented on openprocurement/integrations/edr/auth.py:
```diff
@@ -0,0 +1,30 @@
+# -*- coding: utf-8 -*-
+
+
+def get_local_roles(context):
```
і це також не потрібне

Krokop replied:
get_local_roles потрібно видалити або покрити тестами?

kroman0 replied:
get_local_roles не потрібне оскільки нема контексту для пошуку локальних ролей

Krokop replied:
прибрав

kroman0 commented on openprocurement/integrations/edr/auth.py:
```diff
@@ -0,0 +1,30 @@
+    return groups[0][2:] if groups else 'anonymous'
+
+
+def check_accreditation(request, level):
```
і це

Krokop replied:
прибрав check_accreditation, бо він не використовувався

kroman0 reviewed (changes requested):
Ще трошки для coverage

Krokop committed:
delete check_accreditation

kroman0 commented:
Не покритий тестами https://coveralls.io/builds/10470939/source?filename=openprocurement%2Fintegrations%2Fedr%2Fclient.py#L31 Чи буде це метод використовуватись?

Krokop committed:
remove get_local_roles

Krokop commented:
https://coveralls.io/builds/10470939/source?filename=openprocurement%2Fintegrations%2Fedr%2Fclient.py#L31  - метод потрібен для наступного запиту на EDR, в наступному ПР будуть тести.

kroman0 commented on openprocurement/integrations/edr/utils.py:
```diff
@@ -0,0 +1,167 @@
+        params['ROLE'] = str(errors.request.authenticated_role)
+        if errors.request.params:
+            params['PARAMS'] = str(dict(errors.request.params))
+    if errors.request.matchdict:
```
route не має змінних параметрів, тому request.matchdict можна також забрати

Krokop replied:
прибрав

kroman0 commented on openprocurement/integrations/edr/utils.py:
```diff
@@ -0,0 +1,167 @@
+    params['ROLE'] = str(request.authenticated_role)
+    if request.params:
+        params['PARAMS'] = str(dict(request.params))
+    if request.matchdict:
```
і тут

Krokop replied:
прибрав

kroman0 commented on openprocurement/integrations/edr/utils.py:
```diff
@@ -0,0 +1,167 @@
+    accept = request.headers.get('Accept')
+    jsonp = request.params.get('opt_jsonp')
+    if jsonp and pretty:
+        request.override_renderer = 'prettyjsonp'
```
Ще би мінімальний тест тут - https://coveralls.io/builds/10471928/source?filename=openprocurement%2Fintegrations%2Fedr%2Futils.py#L152
Подібно як тут - https://github.com/openprocurement/openprocurement.api/blob/master/src/openprocurement/api/tests/tender.py#L65

Krokop replied:
Додав

kroman0 reviewed (changes requested):
Ще останні поправки і переходимо до наступних PR

Krokop committed:
add test for opt_jsonp and pretty, remove matchdict

Krokop commented:
Потрібен ревью

kroman0 commented:
Залишилось https://coveralls.io/builds/10472746/source?filename=openprocurement%2Fintegrations%2Fedr%2Futils.py#L152 але загалом 91% coverage.
